editor/editor_node.cpp

```diff
@@ -2308,7 +2308,7 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 			uint32_t current = editor_history.get_current();
 			Object *current_obj = current > 0 ? ObjectDB::get_instance(current) : NULL;
 
-			ERR_FAIL_COND(!current_obj->cast_to<Resource>())
+			ERR_FAIL_COND(!current_obj)
```

Comment:
The checks in this block are not equivalent to the old code. This code only fails if current_obj is Null, while the previous code would also fail if current_obj couldn't cast to a Resource. Any crashes this used to produce will be fixed with #10581 now.